src/ray/gcs/tables.h

```diff
@@ -142,6 +141,71 @@ class Table {
   int64_t subscribe_callback_index_;
 };
 
+/// \class Table
+///
+/// A GCS table where every entry is a single data item.
+template <typename ID, typename Data>
+class Table : private Log<ID, Data> {
+ public:
+  using DataT = typename Log<ID, Data>::DataT;
+  using Callback =
+      std::function<void(AsyncGcsClient *client, const ID &id, const DataT &data)>;
+  /// The callback to call when a Lookup call returns an empty entry.
+  using FailureCallback = std::function<void(AsyncGcsClient *client, const ID &id)>;
+  /// The callback to call when a Subscribe call completes and we are ready to
+  /// request and receive notifications.
+  using SubscriptionCallback = typename Log<ID, Data>::SubscriptionCallback;
+
+  struct CallbackData {
+    ID id;
+    std::shared_ptr<DataT> data;
+    Callback callback;
+    // An optional callback to call for subscription operations, where the
+    // first message is a notification of subscription success.
+    SubscriptionCallback subscription_callback;
+    Log<ID, Data> *log;
+    AsyncGcsClient *client;
+  };
+
+  Table(const std::shared_ptr<RedisContext> &context, AsyncGcsClient *client)
+      : Log<ID, Data>(context, client) {}
```

Comment:
Doesn't `super(...)` work?

Reply:
I'm not clear on how you do that. Can you explain?